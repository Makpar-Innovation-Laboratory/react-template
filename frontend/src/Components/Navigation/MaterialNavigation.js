import React, {useContext} from "react";
import { Box, AppBar, Toolbar, CssBaseline, Typography, makeStyles, useTheme, useMediaQuery, colors, Button, ThemeProvider} from "@material-ui/core";
import {red, blue, common, yellow} from "@material-ui/core/colors"
import { Link } from "react-router-dom";
import DrawerComponent from "./Drawer";
import Logo from '../../Utility/Images/makpar_icon.ico'
import Auth from '../../Utility/Auth'
import {Context} from '../../App'
import {useNavigate} from 'react-router-dom'
import theme from '../../Utility/theme'

const useStyles = makeStyles((theme) => ({
  root:{background: blue[700]},
  navlinks: {
    marginLeft: theme.spacing(5),
    display: "flex justify-content-start",
  },
  logo: {
    flexGrow: "1",
    cursor: "pointer",
    color: common.white
  },
  link: {
    underlineHover:{color: red[500]},
    textDecoration: "none",
    fontSize: "20px",
    marginLeft: theme.spacing(5),
    "&:hover": {
      color: yellow,
      borderBottom: "1px solid white",
    }
  },
  text:{
    color: common.white, 
    children: "link"
  },
  button:{
    containedSecondary:{
      color: blue[50],
    },
  }
}));

function Navbar() {
  const classes = useStyles();
  const theme = useTheme();
  const navigate = useNavigate()
  const context = useContext(Context)
  const isMobile = useMediaQuery(theme.breakpoints.down("md"));
  function handleSubmit(e) {
    e.preventDefault()
    console.log('clicked')
    Auth.deauthenticateUser()
    context.setAuth(false)
    navigate('/Login')
  }
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <AppBar position="static" className={classes.root}>
        <Toolbar>
          <Typography variant="h4" className={classes.logo}>
            Makpar
            <Box
              component="img"
              sx={{
                height: 64,
              }}
              alt="Your logo."
              src={Logo}
            />
          </Typography>
          {isMobile ? (
            <DrawerComponent />
          ) : (
            <div className={classes.navlinks}> 
              <Button variant='text' href="/"><Typography className={classes.text}> Home</Typography></Button>
              <Button variant='text' href="/Mission"><Typography className={classes.text}> Mission</Typography></Button>
              <Button variant='text' href="/PostArchive"><Typography className={classes.text}> Post Archive </Typography></Button>
              <Button variant='text' href="/Profiles"><Typography className={classes.text}> Text </Typography></Button>
              {/* <Link to="/"  className={classes.link} underline='hover'>
                Home
              </Link>
              
              <Link to="/Mission" className={classes.link}>
                Mission
              </Link>
              <Link to="/PostArchive" className={classes.link}>
                Post Archive
              </Link>
              <Link to="/Profiles" className={classes.link}>
                Developer Profiles
              </Link> */}
              
              <Button className={classes.button} variant="contained" color="secondary" onClick={handleSubmit}>Log Out</Button>  
            </div>
            
          )}
        </Toolbar>
      </AppBar>
    </ThemeProvider>
  );
}
export default Navbar;