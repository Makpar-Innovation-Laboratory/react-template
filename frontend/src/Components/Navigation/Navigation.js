import React from 'react'
// import { Context } from "../../App";
import { Link, useLocation } from 'react-router-dom'

export default function Navigation () {
  const location = useLocation()

  return (
    <div className='nav-container background-light d-flex align-items-center py-4'>
      <div className='nav-header px-5'>
        <h1 className='color-primary'>MAKPAR</h1>
      </div>

      <div className='nav-items d-flex justify-content-start align-items-center'>
        <a>
          <div
            className='nav-item color-primary'
            id={location.pathname == '/' ? 'active' : ''}
          >
            <Link to='/' className='color-primary'>
              <h5 className='color-primary'>Home</h5>
            </Link>
          </div>
        </a>
      </div>

      <div className='px-5'>
        <a className=' color-primary'>
          <h5 className='color-primary'>Logout</h5>
        </a>
      </div>
    </div>

  )
}
