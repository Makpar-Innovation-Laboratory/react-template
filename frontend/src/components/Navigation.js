import React, { useState } from "react";
import { NavLink } from "react-router-dom";

const Navigation = () => {
  const [menuExpanded, expandMenu] = useState(false);
  const activeClass = "active-style-navigation";

  return (
    <div className="navigation-container">
      <div id="menu-icon" onClick={() => expandMenu(!menuExpanded)}>&#9776;</div>
      {menuExpanded &&
        <ul id="drop-down-menu" className="usa-sidenav" >
          <li className="usa-sidenav__item" >
            <NavLink id="home-nav-link" exact={true} to="/" activeClassName={activeClass} onClick={() => expandMenu(false)}>Home</NavLink>
          </li>
          <li className="usa-sidenav__item" >
            <NavLink id="application-nav-link" to="/form" activeClassName={activeClass} onClick={() => expandMenu(false)}>Application Form</NavLink>
          </li>
        </ul>
      }
    </div>
  );
};


export default Navigation;
