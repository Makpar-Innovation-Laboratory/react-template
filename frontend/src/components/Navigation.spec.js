import React from "react";
import { isolateComponent } from "isolate-react";
import Navigation from "./Navigation";
import { history } from "../store/configureStore";

describe("JSX Tests :: <Navigation /> Rendering", () => {

  test("Sanity Check Snapshot Rendering", () => {
    const wrapper = isolateComponent(<Navigation />);

    expect(wrapper.debug()).toMatchSnapshot();
  });

  test("Drop-down Menu Shows When Toggled", () => {
    const wrapper = isolateComponent(<Navigation />);
    const toggleBtn = wrapper.findOne("#menu-icon");
    const dropDown = wrapper.findAll("#drop-down-menu");

    expect(dropDown.length).toEqual(0);

    toggleBtn.props.onClick();
    const dropDownAfter = wrapper.findAll("#drop-down-menu");

    expect(dropDownAfter.length).toEqual(1);
  });

  test("Home Navigation Menu Item Correctly Navigates To Home Page", () => {
    const props = ({ location: "/about"});
    const wrapper = isolateComponent(<Navigation {...props} />);
    const toggleBtn = wrapper.findOne("#menu-icon");
    toggleBtn.props.onClick();
    const homeNav = wrapper.findOne("#home-nav-link");
    homeNav.props.onClick();

    expect(history.location.pathname).toEqual("/");
  });

  xtest("Application Button Correctly Navigates To Form Page", () => {
    const props = ({ location: "/about"});
    const wrapper = isolateComponent(<Navigation {...props} />);
    const toggleBtn = wrapper.findOne("#menu-icon");
    toggleBtn.props.onClick();
    const formNav = wrapper.findOne("#application-nav-link");

    formNav.props.onClick();

    expect(history.location.pathname).toEqual("/form");
  });
});
