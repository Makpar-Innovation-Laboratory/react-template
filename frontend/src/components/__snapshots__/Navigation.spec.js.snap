// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX Tests :: <Navigation /> Rendering Sanity Check Snapshot Rendering 1`] = `
"{\\"nodeType\\":\\"html\\",\\"type\\":\\"div\\",\\"name\\":\\"div\\",\\"props\\":{\\"className\\":\\"navigation-container\\",\\"children\\":[{\\"type\\":\\"div\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"id\\":\\"menu-icon\\",\\"children\\":\\"☰\\"},\\"_owner\\":null,\\"_store\\":{}},false]},\\"key\\":\\"null\\"}
 -> {\\"nodeType\\":\\"html\\",\\"type\\":\\"div\\",\\"name\\":\\"div\\",\\"props\\":{\\"id\\":\\"menu-icon\\",\\"children\\":\\"☰\\"},\\"key\\":\\"null\\"}
  -> {\\"nodeType\\":\\"string\\",\\"type\\":\\"☰\\",\\"props\\":{},\\"name\\":\\"\\",\\"key\\":\\"undefined\\"}
 -> {\\"nodeType\\":\\"nothing\\",\\"type\\":\\"false\\",\\"name\\":\\"\\",\\"props\\":{}}"
`;
